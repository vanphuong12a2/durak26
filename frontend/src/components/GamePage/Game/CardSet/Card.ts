export interface Card {
  id: string
  isUp: boolean
}
