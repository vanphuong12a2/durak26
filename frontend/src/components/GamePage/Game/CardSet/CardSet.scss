.card-set {
  display: grid;
  grid-template-columns: repeat(auto-fit,  minmax(0px, max-content)) ;
  z-index: 1;
  padding-right: 50px;
}
