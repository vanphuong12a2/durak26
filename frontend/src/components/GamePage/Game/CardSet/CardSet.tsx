import React from 'react';
import {Card, FaceDownCard, FaceUpCard} from './Card/Card';
import './CardSet.scss';

interface Props {
  cards: Card[]
}

const FaceUpCardSet = (props: Props) => {
  return (
    <div className='card-set'>
      {props.cards.map((card, index) => <FaceUpCard key={index} card={card}/>)}
    </div>
  );
};

const FaceDownCardSet = (props: { numberOfCards: number }) => {
  return (
    <div className='card-set'>
      {
        new Array(props.numberOfCards).fill(0).map((value, index) => <FaceDownCard key={index}/>)
      }
    </div>
  );
};

export {FaceUpCardSet, FaceDownCardSet};
