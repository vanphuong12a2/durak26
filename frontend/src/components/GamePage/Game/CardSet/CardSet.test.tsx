import React from 'react';
import {create} from 'react-test-renderer';
import {FaceDownCardSet, FaceUpCardSet} from './CardSet';

describe('<FaceUpCardSet />', () => {

  describe('Snapshots', () => {
    it('renders FaceUpCardSet correctly', () => {
      const component = create(<FaceUpCardSet cards={[]}/>);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('renders FaceDownCardSet correctly', () => {
      const component = create(<FaceDownCardSet numberOfCards={4}/>);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

});
