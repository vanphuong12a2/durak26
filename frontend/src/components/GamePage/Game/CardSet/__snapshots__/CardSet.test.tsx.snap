// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FaceUpCardSet /> Snapshots renders FaceDownCardSet correctly 1`] = `
<div
  className="card-set"
>
  <img
    className="card"
    src="card_back.svg"
  />
  <img
    className="card"
    src="card_back.svg"
  />
  <img
    className="card"
    src="card_back.svg"
  />
  <img
    className="card"
    src="card_back.svg"
  />
</div>
`;

exports[`<FaceUpCardSet /> Snapshots renders FaceUpCardSet correctly 1`] = `
<div
  className="card-set"
/>
`;
